buildspecs/release-config.yml

```diff
@@ -56,7 +56,24 @@ phases:
           }
         }
         EOF
+      
+      # Copying over the existing agentVersionV2-<branch>.json file 
+      - aws s3 cp ${RESULTS_BUCKET_URI}/agentVersionV2/agentVersionV2-${GITHUB_SOURCE_BRANCH_NAME}.json ./agentVersionV2-${GITHUB_SOURCE_BRANCH_NAME}.json
+      # Grabbing the old release agent version (will be the new current agent version)
+      - CURR_VER=$(jq -r '.agentReleaseVersion' agentVersionV2-${GITHUB_SOURCE_BRANCH_NAME}.json)
+      # Updating the agentVersionV2-<branch>.json file with new current and release agent versions
+      - cat <<< $(jq '.agentReleaseVersion = '\"$AGENT_VERSION\"' | .agentCurrentVersion = '\"$CURR_VER\"'' agentVersionV2-${GITHUB_SOURCE_BRANCH_NAME}.json) > agentVersionV2-${GITHUB_SOURCE_BRANCH_NAME}-COPY.json
+      # Reformatting the file
+      - jq . agentVersionV2-${GITHUB_SOURCE_BRANCH_NAME}-COPY.json > agentVersionV2-${GITHUB_SOURCE_BRANCH_NAME}.json
+
+      # Checking if there's no new agent version
```

Comment:
Yeah, I agree. I added this check just so the buildspec script would be a bit smart/know not to have an agent version pointing to itself as the next version. Per offline, I think we can remove this check based on our mechanism on what to do when an agent version needs to be "rolled back".

Reply:
Thanks, I see that this is a sanity check for when current version = release version due to some misconfiguration. We may keep it as well, as it covers one scenario :) Although human eyes would to confirm validity of the file might still be necessary.

Reply:
Hm, maybe in the case where the release agent versions remain unchanged we don't do any modification. That might be a better way to achieve idempotence.

Reply:
> 

Yep, completely agree with that 🙂 . Perhaps we'll need to output the config file with the updated changes so that we don't need to manually download the files to check?